lib/csetgrp.gi

```diff
@@ -667,11 +667,11 @@ function(a,b)
 local c;
   if ActingDomain(a)<>ActingDomain(b) then TryNextMethod();fi;
   if not IsBiCoset(a) then # product does not require b to be bicoset
-    TryNextMethod();
+    ErrorNoReturn("right cosets can only be multiplied if the left operand is a bicoset");
```

Comment:
Making this an error will hard block any future attempt to define multiplication of ordinary cosets (e.g. as multiplication of sets) in a more general way through other methods. Is this deliberate?

Reply:
This was quite deliberate, i.e., I thought quite a while about whether to change this into an error; or whether to instead change the `Inverse` method to use `TryNextMethod()` instead of `Error` -- in any case, I think they should match.

However, in the end I think a clear helpful error message is far better than a "method not found", just to support a hypothetical future extension which may never come. Besides, we actually don't stop anybody from adding one: they just have to rank their method higher than ours (and make sure to stay compatibly / invoke `TryNextMethod()` if needed). Finally, we can of course always change the error into something else if it ever becomes necessary.